core/src/main/scala/com/databricks/labs/remorph/graph/lineage.scala

```diff
@@ -0,0 +1,41 @@
+package com.databricks.labs.remorph.graph
+
+import scala.collection.mutable
+
+case object Action extends Enumeration {
+  type Operation = Value
+  val Read, Write, Update, Delete, Merge = Value
+}
+
+case class Node(catalog: Option[String], schema: String, table: String)
+
+case class Edge(from: Node, operator: Action.Operation, to: Node)
+
```

Comment:
why edge needed? We can have Node and from: Node, to: Node can and  operator: Action.Operation can be the properties of the Node.